src/Middleware/RequestThrottling/src/RequestThrottlingEventSource.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.RequestThrottling
+{
+    internal sealed class RequestThrottlingEventSource : EventSource
+    {
+        public static readonly RequestThrottlingEventSource Log = new RequestThrottlingEventSource();
+        private static QueueFrame _cachedNonTimerResult = new QueueFrame
+        {
+            _parent = Log
+        };
+
+        private PollingCounter _rejectedRequestsCounter;   // incrementing polling counter
+        private PollingCounter _queueLengthCounter;        // polling counter
+        private EventCounter _queueDuration;     // event counter average
+
+        internal long _rejectedRequests;
+        internal int _queueLength;
+
+    
+        internal RequestThrottlingEventSource()
+            : base("Microsoft.AspNetCore.RequestThrottling")
+        {
+        }
+
+        // Used for testing
+        internal RequestThrottlingEventSource(string eventSourceName)
+            : base(eventSourceName)
+        {
+        }
+
+        [Event(1, Level = EventLevel.Warning)]
+        public void RequestRejected()
```

Comment:
Should we include the request's `HttpContext.TraceIdentifier` value here? cc @halter73 @davidfowl ?

It's not relevant to the counter, which is just a number, but would be useful data if you were actually capturing the raw event. We don't really have access to logging scopes here, so we can't rely on that.

Reply:
@DylanDmitri get @halter73 and/or @shirhatti to help you capture the raw events as well so you can see how that works and the data associated with them. You can use `dotnet-trace`, by enabling the `Microsoft.AspNetCore.RequestThrottling` provider, and view the data in PerfView:

1. Install using: `dotnet tool install --global dotnet-trace --version 1.0.4-preview6.19311.1`
1. Launch the app, get the process ID
1. Start collecting: `dotnet trace collect --providers Microsoft.AspNetCore.RequestThrottling -p [pid]`
1. Open `trace.netperf` in [PerfView](http://github.com/Microsoft/PerfView)

Reply:
We don't really have a good way to look at these events and their arguments, we're mainly using the EventSource as a means to an end (counters), so I wouldn't spend too much time on the structure of the events as we're inconsistent all over the place.

Reply:
I've ran the tool and watched the counters; they function as expected.